maven-core/src/main/java/org/apache/maven/graph/DefaultProjectDependencyGraph.java

```diff
@@ -62,38 +62,30 @@
      * @throws CycleDetectedException
      */
     public DefaultProjectDependencyGraph( Collection<MavenProject> projects )
-        throws CycleDetectedException, DuplicateProjectException
+            throws CycleDetectedException, DuplicateProjectException
     {
-        super();
-        this.allProjects = Collections.unmodifiableList( new ArrayList<>( projects ) );
-        this.sorter = new ProjectSorter( projects );
-        this.order = new HashMap<>();
-        this.projects = new HashMap<>();
-        List<MavenProject> sorted = this.sorter.getSortedProjects();
-        for ( int index = 0; index < sorted.size(); index++ )
-        {
-            MavenProject project = sorted.get( index );
-            String id = ProjectSorter.getId( project );
-            this.projects.put( id, project );
-            this.order.put( project, index );
-        }
+        init( projects, projects );
     }
 
     /**
      * Creates a new project dependency graph based on the specified projects.
      *
      * @param allProjects All collected projects.
-     * @param projects The projects to create the dependency graph with.
-     *
+     * @param projects    The projects to create the dependency graph with.
      * @throws DuplicateProjectException
      * @throws CycleDetectedException
      * @since 3.5.0
      */
     public DefaultProjectDependencyGraph( final List<MavenProject> allProjects,
                                           final Collection<MavenProject> projects )
-        throws CycleDetectedException, DuplicateProjectException
+            throws CycleDetectedException, DuplicateProjectException
+    {
+        init( allProjects, projects );
+    }
+
+    private void init( Collection<MavenProject> allProjects, final Collection<MavenProject> projects )
```

Comment:
Why not remove this new method, and in `DefaultProjectDependencyGraph( Collection<MavenProject>  )` use this Ctor, and bring final fields back?

Reply:
I feel the same. The init isn't that complex that it would require a separate method. Final is a nice thing.

Reply:
Yeah I think it was too late...this morning...If I change the signature of one of the Ctors...this can simply being done...